### ISSUE TITLE
Master Alchemist 20% increased effect of magic ultility Flasks is not caluclated

### ISSUE THREAD

**Issue by JonasHaba** [PathOfBuildingCommunity/PathOfBuilding#4868]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

The rarity value in the calculation should go up.
Because the "Abecedarian's Gold Flask of the Order" Flask should get 20% increased effect. 

### What is the actual behaviour/value?

The Flask is getting the 20% effect, but the rarity in the calcs isn't going up.

### How to reproduce the issue

Click on "Master Alchemist" go to calc. 
Look at the number.
Do it again.
Look at the number.

Rarity without Master Alchemist: 635%
Rarity with Master Alchemist: 635%

### Character build code

```shell
https://pastebin.com/j2LsXMS1
```

**Comment by QuickStick123** [comment]
This is not an issue 
With
![image](https://user-images.githubusercontent.com/31533893/184998025-fe62a733-3c12-42a3-b0f4-a865bbba017e.png)
Without
![image](https://user-images.githubusercontent.com/31533893/184998054-eefed616-9b86-4f77-b5d6-1ef3b0ddffc9.png)

**Comment by afishyate** [comment]
I've also had issues with this node since implemented.
-It randomly has no effect on all, or sometimes one or two particular magic utility flasks
-It doesn't seem to happen at all on newly created POB profiles
-It seems to happen often on big/old POB profiles, with 5+ tree/gem/gear sets
-Sometimes these profiles work normally after restarting POB, but then become dysfunctional after some time

**Comment by QuickStick123** [comment]
Can you provide a build code that this occurs on?

**Comment by afishyate** [comment]
https://pastebin.com/MErDA0AC
I have:
Jade + Evasion
Silver + Resistance
Quicksilver + Movespeed

Each time I open the build, and re-allocate the Master Alchemist, random sidebar stats go unchanged, or none change.
e.g., Movement speed increases, but resists and evasion do not.

**Comment by QuickStick123** [comment]
There is something odd going on here re openning.

**Comment by QuickStick123** [comment]
It seems certain mods arn't getting scaled correctly by the magic thing namely movement speed flask suffix, jade flask prefix and evasion suffix by magic flask effect.

**Comment by afishyate** [comment]
Correct. The strangest part is how random it is. It can be changed by:
-Re-opening the profile
-Making several changes to other parts of the profile
-Making a new identical flask seems to 'fix' the broken/unaffected stats, but sometimes doesn't work

Feels like each time I open the profile, I get a random bug seed lol.

**Comment by afishyate** [comment]
Just discovered, that having the Eternal Mana Flask active is what messes things up. Tried a new profile, all my old ones, a new one in 2.20.1. Every time, activating the mana flask creates the previously described bug: MA's 20% flask effect is removed from some or all of the stats affected by magic utility flask mods.

**Comment by afishyate** [comment]
https://streamable.com/bfimq0
I captured a video here to show the exact issue. After recording, I tested with unique/normal Life flasks, and the bug is the same.

My last post still stands, that a magic mana flask alone was changing stats. Here (in this bug seed? lol) they don't seem to do anything without a unique mana flask also active.

But as shown, switching each slot creates a variety of results. In former tests, it even seems to sometimes to cancel out the removed 20% that the other mana flask imposed.
